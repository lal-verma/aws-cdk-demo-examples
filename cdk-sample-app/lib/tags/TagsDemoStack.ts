import * as cdk from 'aws-cdk-lib';
import { Construct } from 'constructs';
import * as dynamodb from 'aws-cdk-lib/aws-dynamodb';
import * as lambda from 'aws-cdk-lib/aws-lambda';
import path = require('path');

/**
 * TagsDemoStack_1
 *  - todo-db
 *  - todo-service
 * TagsDemoStack_2
 *  - todo-db
 *  - todo-service
 */

export class TagsDemoStack_1 extends cdk.Stack {
  constructor(scope: Construct, id: string, props?: cdk.StackProps) {

    super(scope, id, props);

    new dynamodb.Table(this, 'todo-db', {
        partitionKey: { name: 'id', type: dynamodb.AttributeType.STRING },
    });


    new lambda.Function(this, 'todo-service', {
        runtime: lambda.Runtime.NODEJS_18_X,
        handler: 'todo-service.handler',
        code: lambda.Code.fromAsset(path.join(__dirname, '../../../sample-lambda-functions')),
    });
  
    cdk.Tags.of(this).add('microservice', 'todo-service');
  }
}

/**
 * Following Stack provisions the AWS resources for Microservice A Database - DynamoDB
 */
export class TagsDemoStack_2 extends cdk.Stack {
  constructor(scope: Construct, id: string, props?: cdk.StackProps) {

    super(scope, id, props);

    new dynamodb.TableV2(this, 'todo-db', {
        partitionKey: { name: 'id', type: dynamodb.AttributeType.STRING },
    });


    const todo_service_function = new lambda.Function(this, 'todo-service', {
        runtime: lambda.Runtime.NODEJS_18_X,
        handler: 'todo-service.handler',
        code: lambda.Code.fromAsset(path.join(__dirname, '../../../sample-lambda-functions')),
        
    }); 

    cdk.Tags.of(todo_service_function).add('microservice', 'todo-service-v2');

  }
}
