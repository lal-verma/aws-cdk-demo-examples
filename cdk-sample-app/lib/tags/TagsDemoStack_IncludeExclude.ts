import * as cdk from 'aws-cdk-lib';
import { Construct } from 'constructs';
import * as lambda from 'aws-cdk-lib/aws-lambda';
import * as dynamodb from 'aws-cdk-lib/aws-dynamodb';


import path = require('path');
/**
 * TagsDemoStack_1 (exclude)
 *  -todo-service
 *  -todo-db * 
 * TagsDemoStack_2 (include)
 *  - todo-service
 *  - todo-db

 */

export class TagsDemoStack_1 extends cdk.Stack {
  /**
   *
   * @param {Construct} scope
   * @param {string} id
   * @param {StackProps=} props
   */
  constructor(scope: Construct, id: string, props: cdk.StackProps) {
    super(scope, id, props);


    const todo_service_function =new lambda.Function(this, 'todo-service', {
        runtime: lambda.Runtime.NODEJS_18_X,
        handler: 'todo-service.handler',
        code: lambda.Code.fromAsset(path.join(__dirname, '../../../sample-lambda-functions')),
        
    });

    const todo_db = new dynamodb.Table(this, 'todo-db', {
      partitionKey: { name: 'id', type: dynamodb.AttributeType.STRING },
      removalPolicy: cdk.RemovalPolicy.DESTROY,
    });
 
    cdk.Tags.of(this).add('microservice', 'todo-service', {
      excludeResourceTypes: ['AWS::DynamoDB::Table'],
    });
  
  }
}

export class TagsDemoStack extends cdk.Stack {

  constructor(scope: Construct, id: string, props: cdk.StackProps) {
    super(scope, id, props);


    cdk.Tags.of(this).add('microservice', 'todo-service', {
      excludeResourceTypes: ['AWS::DynamoDB::Table']
    });

    //other code below
  
  }
}