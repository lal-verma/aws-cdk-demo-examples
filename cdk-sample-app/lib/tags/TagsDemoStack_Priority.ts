import * as cdk from 'aws-cdk-lib';
import { Construct } from 'constructs';
import * as lambda from 'aws-cdk-lib/aws-lambda';
import * as dynamodb from 'aws-cdk-lib/aws-dynamodb';
import path = require('path');

/**
 * TagsDemoStack
 *  -todo-service
 *  -todo-db
 */


export class TagsDemoStack extends cdk.Stack {
  /**
   *
   * @param {Construct} scope
   * @param {string} id
   * @param {StackProps=} props
   */
  constructor(scope: Construct, id: string, props: cdk.StackProps) {
    super(scope, id, props);


    const todo_service =new lambda.Function(this, 'todo-service', {
        runtime: lambda.Runtime.NODEJS_18_X,
        handler: 'todo-service.handler',
        code: lambda.Code.fromAsset(path.join(__dirname, '../../../sample-lambda-functions')),
        
    });

    new dynamodb.TableV2(this, 'todo-db', {
      partitionKey: { name: 'id', type: dynamodb.AttributeType.STRING },
    });

    cdk.Tags.of(this).add('microservice', 'todo-service');

    cdk.Tags.of(todo_service).add('microservice', 'todo-service-v2',{priority: 300});    
  
  }
}