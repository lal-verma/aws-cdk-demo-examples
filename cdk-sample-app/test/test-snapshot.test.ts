import * as cdk from 'aws-cdk-lib';
import { Template } from 'aws-cdk-lib/assertions';
import { TagsDemoStack_1, TagsDemoStack_2 } from '../lib/tags/TagsDemoStack';


//demo: Match.objectEquals
test('snapshot test', () => {
  const app = new cdk.App();
//  const stack = new TagsDemoStack_1(app, 'TagsDemoStack-test');
  const stack = new TagsDemoStack_2(app, 'TagsDemoStack-test');
  const template = Template.fromStack(stack);

  expect(template).toMatchSnapshot('snapshot-stack-1');
});
