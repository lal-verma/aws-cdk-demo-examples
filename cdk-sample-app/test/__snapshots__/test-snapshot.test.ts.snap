// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test: snapshot-stack-1 1`] = `
{
  "Parameters": {
    "BootstrapVersion": {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": {
    "tododbA0F8A3AC": {
      "DeletionPolicy": "Retain",
      "Properties": {
        "AttributeDefinitions": [
          {
            "AttributeName": "id",
            "AttributeType": "S",
          },
        ],
        "BillingMode": "PAY_PER_REQUEST",
        "KeySchema": [
          {
            "AttributeName": "id",
            "KeyType": "HASH",
          },
        ],
        "Replicas": [
          {
            "Region": {
              "Ref": "AWS::Region",
            },
          },
        ],
      },
      "Type": "AWS::DynamoDB::GlobalTable",
      "UpdateReplacePolicy": "Retain",
    },
    "todoservice099F38C7": {
      "DependsOn": [
        "todoserviceServiceRole296BAF91",
      ],
      "Properties": {
        "Code": {
          "S3Bucket": {
            "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region}",
          },
          "S3Key": "5b4ccb089a030833daa6a5eea6a74fd72146bf3ddb1edef9cabcd1fa1f54548a.zip",
        },
        "Handler": "todo-service.handler",
        "Role": {
          "Fn::GetAtt": [
            "todoserviceServiceRole296BAF91",
            "Arn",
          ],
        },
        "Runtime": "nodejs18.x",
        "Tags": [
          {
            "Key": "microservice",
            "Value": "todo-service-v2",
          },
        ],
      },
      "Type": "AWS::Lambda::Function",
    },
    "todoserviceServiceRole296BAF91": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": [
          {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
        "Tags": [
          {
            "Key": "microservice",
            "Value": "todo-service-v2",
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
  },
  "Rules": {
    "CheckBootstrapVersion": {
      "Assertions": [
        {
          "Assert": {
            "Fn::Not": [
              {
                "Fn::Contains": [
                  [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;
